packages/base-map/src/index.story.tsx

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable react/button-has-type */
+import React, { useRef } from "react";
+import {
+  AttributionControl,
+  MapboxMap,
+  NavigationControl,
+  Popup,
+  ScaleControl
+} from "react-map-gl";
+import { action } from "@storybook/addon-actions";
+
+import AllVehiclesOverlay from "../__mocks__/AllVehicles";
+import ContextMenuDemo from "../__mocks__/ContextMenuDemo";
+import BaseMap, { MarkerWithPopup, LayerWrapper } from ".";
+
+import "maplibre-gl/dist/maplibre-gl.css";
+
+const center: [number, number] = [45.522862, -122.667837];
+
+export default {
+  args: { center, forceMaxHeight: true },
+  component: BaseMap,
+  title: "BaseMap"
+};
+
+// eslint-disable-next-line react/jsx-props-no-spreading
+const Template = args => <BaseMap {...args} />;
+
+const samplePopup = (
+  <div>
+    <h1>Popup Title</h1>
+    <p>
+      Sample <span style={{ color: "purple" }}>popup</span> content.
+    </p>
+  </div>
+);
+
+const sampleMarkers = (
+  <MarkerWithPopup
+    tooltipContents={samplePopup}
+    position={center}
+  ></MarkerWithPopup>
+);
+
+const onClick = action("onClick");
+const onContextMenu = action("onContextMenu");
+const onViewportChanged = action("onViewportChanged");
+const a11yOverrideParameters = {
+  a11y: { config: { rules: [{ id: "color-contrast", reviewOnFail: true }] } }
+};
+
+export const clickAndViewportchangedEvents = Template.bind({});
+clickAndViewportchangedEvents.args = {
+  center,
+  forceMaxHeight: true,
+  onClick,
+  onContextMenu,
+  onViewportChanged
+};
+
+export const zoomed = Template.bind({});
+zoomed.args = {
+  center,
+  zoom: 17,
+  forceMaxHeight: true
+};
+
+export const clickToSetBounds = () => {
+  const ref = useRef<MapboxMap>();
+  const bbox: [number, number, number, number] = [-79, 43, -73, 45];
+
+  return (
+    <div>
+      <button
+        onClick={
+          () =>
+            ref.current?.fitBounds(bbox, {
+              duration: 300,
+              padding: {
+                bottom: 25,
+                left: 15,
+                right: 5,
+                top: 10
+              }
+            })
+          // eslint-disable-next-line react/jsx-curly-newline
+        }
+      >
+        Set the bounds
+      </button>
+      <BaseMap passedRef={ref} center={center} forceMaxHeight zoom={17} />
```

Comment:
I would try not to pass a ref. Instead, move `<MapProvider>` to the storybook level (similar to where we used `<IntlProvider>`) and use hook `useMap` to access the underlying map object. See https://visgl.github.io/react-map-gl/docs/api-reference/use-map.

Reply:
Surround the `<div>` or replace it with a `<MapProvider>`, give an `id` to the map (see also my comment about passing in an `id` prop), and you will be able to use `useMap`, so we can get rid of the `passedRef` prop.